blocks/bricks/wrappers.py

```diff
@@ -34,3 +35,35 @@ def apply(self, input_):
         else:
             output = self.application_method(input_)
         return output
+
+
+class WithAxesSwapped(Brick):
+    """Wraps an application around dimshuffles.
+
+    Parameters
+    ----------
+    application_method : callable
+        Some brick's :meth:`apply` method
```

Comment:
It might also be a good idea to specify that it expects the application to have exactly one input and one output.